### DISCUSSION TITLE
How do I build typing extensions 4.3.0 locally

### DISCUSSION THREAD

**Discussion by Ganeshkumhar1** [python/typing#1241] [Q&A]
can someone please tell me the steps to build typing extensions 4.3.0 locally using source tar.gz

**Comment by hauntsaninja** [comment]
Just use tar to extract it and use `python -m build .` https://github.com/pypa/build (could also use pip or flit)

It's a pure Python package so there isn't much going on :-)

**Reply by Ganeshkumhar1** [reply]
I have extracted the tar file - 
typing_extensions-4.3.0/CHANGELOG.md
typing_extensions-4.3.0/LICENSE
typing_extensions-4.3.0/README.md
typing_extensions-4.3.0/pyproject.toml
typing_extensions-4.3.0/src/test_typing_extensions.py
typing_extensions-4.3.0/src/typing_extensions.py
typing_extensions-4.3.0/PKG-INFO

How do I build locally using pip. Please suggest.  

bash-4.4$ python -m pip install pyproject.toml
Requirement already satisfied: pyproject.toml in /scratch/gkumhar/view_storage/gkumhar_newpy/python/lib/python3.10/site-packages (0.0.10)
WARNING: Retrying (Retry(total=4, connect=None, read=None, redirect=None, status=None)) after connection broken by 'NewConnectionError('<pip._vendor.urllib3.connection.HTTPSConnection object at 0x7f39191ca7d0>: Failed to establish a new connection: [Errno 101] Network is unreachable')': /simple/toml/



it is trying to connect to internet . which I do not want.

**Reply by hmc-cs-mdrissi** [reply]
You either want —find-links or to use file path to pass pip local directory. There’s nothing special about typing-extensions. 

https://pip.pypa.io/en/stable/cli/pip_install/?highlight=find-links#examples Look at example number 8/10. Or do pip install . from inside directory with typing-extensions.
